File: metodo_bucket.java
//metodo de ordenamiento bucket sort
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
public class metodo_bucket {
    public static void bucketSort(int[] array, int numBuckets) {
        if (array.length == 0) {
            return;
        }

        // Encontrar el valor máximo y mínimo en el array
        int minValue = array[0];
        int maxValue = array[0];
        for (int value : array) {
            if (value < minValue) {
                minValue = value;
            } else if (value > maxValue) {
                maxValue = value;
            }
        }

        // Crear los buckets
        List<List<Integer>> buckets = new ArrayList<>(numBuckets);
        for (int i = 0; i < numBuckets; i++) {
            buckets.add(new ArrayList<>());
        }

        // Distribuir los elementos en los buckets
        for (int value : array) {
            int bucketIndex = (int) ((value - minValue) * numBuckets / (maxValue - minValue + 1));
            buckets.get(bucketIndex).add(value);
        }

        // Ordenar cada bucket y concatenar los resultados
        int index = 0;
        for (List<Integer> bucket : buckets) {
            Collections.sort(bucket);
            for (int value : bucket) {
                array[index++] = value;
            }
        }
    }
    // Ejemplo de uso
    public static void main(String[] args) {
        int[] array = {34, 2, 10, 6, 7, 5, 3, 1, 9, 8, 4};
        int numBuckets = 5;

        System.out.println("Array original:");
        for (int value : array) {
            System.out.print(value + " ");
        }
        System.out.println();

        bucketSort(array, numBuckets);

        System.out.println("Array ordenado:");
        for (int value : array) {
            System.out.print(value + " ");
        }
    }
    
}

File: metodo_bucket.py
# metodo de ordenamiento bucket sort
def bucket_sort(arr):
    if len(arr) == 0:
        return arr

    # Crear los buckets
    min_value = min(arr)
    max_value = max(arr)
    bucket_count = len(arr)
    bucket_size = (max_value - min_value) / bucket_count
    buckets = [[] for _ in range(bucket_count)]

    # Distribuir los elementos en los buckets
    for num in arr:
        index = int((num - min_value) / bucket_size)
        if index == bucket_count:  # Manejar el caso del valor máximo
            index -= 1
        buckets[index].append(num)

    # Ordenar cada bucket y concatenar los resultados
    sorted_arr = []
    for bucket in buckets:
        sorted_arr.extend(sorted(bucket))

    return sorted_arr

# Ejemplo de uso
if __name__ == "__main__":
    data = [0.78, 0.17, 0.39, 0.26, 0.72, 0.94, 0.21, 0.55, 0.43, 0.88]
    sorted_data = bucket_sort(data)
    print("Array original:", data)
    print("Array ordenado:", sorted_data)